pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```
@@ -388,8 +401,66 @@ private void addExtraFiles(File m2Repo) {
+        }
+        log.debug("Number of collected artifacts for the Offline manifest: {}", artifactsToListRaw.size());
+
+        artifactsToListRaw.removeIf(artifact -> isArtifactExcluded(artifact.getGapv()));
+        log.debug("Number of collected artifacts after exclusion: {}", artifactsToListRaw.size());
+        Map<String, ArtifactWrapper> artifactsToList = new HashMap<>();
+        for (ArtifactWrapper artifact : artifactsToListRaw) {
+            artifactsToList.put(artifact.getGapv(), artifact);
+        }
+        log.info("Number of collected artifacts without duplicates: {}", artifactsToList.size());
+
+        String filename = releasePath + getGenerationData().getOfflinerManifest();
+        try (PrintWriter file = new PrintWriter(filename)){
+            for (Map.Entry<String, ArtifactWrapper> artifactEntry : artifactsToList.entrySet()) {
+                ArtifactWrapper artifact = artifactEntry.getValue();
+                // TODO: Remove the check, when https://projects.engineering.redhat.com/browse/NCL-6079 is done
+                if (artifact.getRepositoryType().equals(RepositoryType.MAVEN)) {
+                    GAV gav = artifact.toGAV();
+                    String offlinerString = String
+                            .format("%s,%s/%s", artifact.getSha256(), gav.toVersionPath(), gav.toFileName());
+                    file.println(offlinerString);
+                }
+
+            }
+            List<GAV> extraGavs = generationData.getExternalAdditionalArtifacts()
+                    .stream()
+                    .map(GAV::fromColonSeparatedGAPV)
+                    .collect(Collectors.toList());
+            for (GAV extraGav : extraGavs) {
+                String offlinerString = String.format("%s/%s", extraGav.toVersionPath(), extraGav.toFileName());
+                file.println(offlinerString);
+            }
+
+        } catch (Exception e) {
+            log.error("Error generating the Offline manifest", e);
```

Isn't this silently swallowing potential issues?

---

It seems that the use of `catch (Exception e)`  is quite common across the codebase to ignore particular exceptions, it's not clear from the code which ones are even possible.

---

Sure, but doesn't mean it should be propagated :-)

---

Shouldn't something like `FileNotFoundException` be handled by the caller, not just logged in the method itself?